src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -129,8 +127,22 @@ class TaskReplaceActor(
       instancesStarted -= 1
       launchInstances().pipeTo(self)
 
+    // An old instance terminated out of band and was not yet chosen to be decommissioned or stopped
+    // the instance will be rescheduled with the new version so we don't have to do anything
+    case InstanceChanged(id, _, `pathId`, condition, instance) if oldInstanceIds(id) && instance.state.goal == Goal.Running =>
```

Comment:
I'd really like to have a discussion about this. I'm having a difficult time connecting the comment to the if statement. It says:

> An old instance terminated out of band 

But the if clause merely checks that it's

* One of the old instance ids
* Goal is still running

Could the condition have transitioned to unreachable?

Reply:
yeah this is a bug, we were discussing it with Matthias this morning. This condition should contain `considerTerminal` as well...